hack/.ci/run-e2e-gke.sh

```diff
@@ -147,7 +148,7 @@ kubectl wait --for condition=established crd/nodeconfigs.scylla.scylladb.com
 if [[ "${SO_DISABLE_NODECONFIG}" == "true" ]]; then
```

Comment:
It might be good to add `OR` here on `SO_NODECONFIG_PATH` being empty, so we can eventually drop the `SO_DISABLE_NODECONFIG` when we switch CI

Reply:
added `or` and changed how default value is assigned when explicit empty string is passed by.